### ISSUE TITLE
`foo && foo['bar']` errors with autocorrect

### ISSUE THREAD

**Issue by ragesoss** [bbatsov/rubocop#4809]
## Expected behavior

Rubocop should autocorrect without error when applied to a safe navigation violation that involves hash access, or the safe navigation rule should not be applied to special syntax like calls to `:[]`.

## Actual behavior

If I run `rubocop -a` on a file that includes syntax like `foo && foo['bar']`, it errors. For exampe:

```
sage@ragesoss-desktop:~/WikiEduDashboard$ rubocop -a lib/training/wiki_slide_parser.rb
Inspecting 1 file


0 files inspected, no offenses detected
undefined method `begin' for nil:NilClass
/home/sage/.rvm/gems/ruby-2.4.2/gems/parser-2.4.0.0/lib/parser/source/rewriter.rb:67:in `insert_before'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/corrector.rb:78:in `insert_before'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/style/safe_navigation.rb:99:in `block in autocorrect'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/corrector.rb:56:in `block (2 levels) in rewrite'
/home/sage/.rvm/gems/ruby-2.4.2/gems/parser-2.4.0.0/lib/parser/source/rewriter.rb:194:in `transaction'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/corrector.rb:55:in `block in rewrite'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/corrector.rb:53:in `each'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/corrector.rb:53:in `rewrite'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/team.rb:126:in `autocorrect_all_cops'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/team.rb:70:in `autocorrect'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/team.rb:98:in `block in offenses'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/team.rb:115:in `investigate'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/team.rb:94:in `offenses'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cop/team.rb:44:in `inspect_file'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:258:in `inspect_file'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:205:in `block in do_inspection_loop'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:237:in `block in iterate_until_no_changes'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:230:in `loop'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:230:in `iterate_until_no_changes'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:201:in `do_inspection_loop'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:111:in `block in file_offenses'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:121:in `file_offense_cache'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:109:in `file_offenses'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:100:in `process_file'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:78:in `block in each_inspected_file'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:75:in `each'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:75:in `reduce'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:75:in `each_inspected_file'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:67:in `inspect_files'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/runner.rb:39:in `run'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cli.rb:86:in `execute_runner'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/lib/rubocop/cli.rb:32:in `run'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/bin/rubocop:13:in `block in <top (required)>'
/home/sage/.rvm/rubies/ruby-2.4.2/lib/ruby/2.4.0/benchmark.rb:308:in `realtime'
/home/sage/.rvm/gems/ruby-2.4.2/gems/rubocop-0.50.0/bin/rubocop:12:in `<top (required)>'
/home/sage/.rvm/gems/ruby-2.4.2/bin/rubocop:23:in `load'
/home/sage/.rvm/gems/ruby-2.4.2/bin/rubocop:23:in `<main>'
/home/sage/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `eval'
/home/sage/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `<main>'
```

## Steps to reproduce the problem

Run `rubocop -a` on a file that includes this line:

```ruby
foo && foo['bar']
```

It's trying to autocorrect the safe navigation violation.

## RuboCop version
```
$ rubocop -V
0.50.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-linux)
```

(Same behavior with 2.3.1)

**Comment by ragesoss** [comment]
Separately from the error, I think this rule should not be applied for hash access.

In my opinion,
```ruby
foo && foo[:bar]
```
is preferable to
```ruby
foo&.[] :bar
```

**Comment by Drenmi** [comment]
Duplicate of #4731, #4741, #4760, #4766, #4771, #4803, #4808. 🙂 Fixed in #4749. Bugfix release is coming soon. 🙂
